source/common/http/header_map_impl.cc

```diff
@@ -70,7 +78,8 @@ void HeaderString::append(const char* data, uint32_t size) {
     // Rather than be too clever and optimize this uncommon case, we dynamically
     // allocate and copy.
     type_ = Type::Dynamic;
-    dynamic_capacity_ = (string_length_ + size) * 2;
+    dynamic_capacity_ =
+        std::max(MinDynamicCapacity, static_cast<size_t>((string_length_ + size) * 2));
```

Comment:
You use `size_t` here, but below the same calculation below uses `static_cast<uint64_t>`. They should be the same, but it might be good to be consistent here, probably standardizing on `size_t`.